src/runtime-rs/crates/resource/src/network/network_info/network_info_from_dan.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2019-2023 Alibaba Cloud
+// Copyright (c) 2019-2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use agent::{ARPNeighbor, IPAddress, IPFamily, Interface, Route};
+use anyhow::Result;
+use async_trait::async_trait;
+use netlink_packet_route::IFF_NOARP;
+
+use super::NetworkInfo;
+use crate::network::dan::DanDevice;
+use crate::network::utils::address::parse_ip_cidr;
+
+/// NetworkInfoFromDan is responsible for converting network info in JSON
+/// to agent's network info.
+#[derive(Debug)]
+pub(crate) struct NetworkInfoFromDan {
+    interface: Interface,
+    routes: Vec<Route>,
+    neighs: Vec<ARPNeighbor>,
+}
+
+impl NetworkInfoFromDan {
+    #[allow(dead_code)]
```

Comment:
I think you could remove this as it's been called

Reply:
Fixed.